projects/storefrontlib/src/cms-components/myaccount/notification-preference/notification-preference.component.ts

```diff
@@ -0,0 +1,51 @@
+import { Component, OnInit } from '@angular/core';
+import {
+  NotificationPreference,
+  UserNotificationPreferenceService,
+} from '@spartacus/core';
+import { Observable, combineLatest } from 'rxjs';
+import { map, tap } from 'rxjs/operators';
+
+@Component({
+  selector: 'cx-notification-preference',
+  templateUrl: './notification-preference.component.html',
+})
+export class NotificationPreferenceComponent implements OnInit {
+  preferences$: Observable<NotificationPreference[]>;
+  isLoading$: Observable<boolean>;
+  preferences: NotificationPreference[] = [];
```

Comment:
Do you need this?

Reply:
Yes, it is used in updatePrefercnec(...).